### PR TITLE
Class structured

### PR DESCRIPTION
**Pull request by bsun94**
Updated code structures in both corpusBuilder and RousseauBot to be set in classes. Updated readme doc as well for better formatting.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,9 +1,13 @@
...
 
-The RousseauBot script is dependent on a JSON file containing the credentials to RousseauBot's Twitter account, which it reads and uses in accessing the account (API/access keys/tokens). The JSON file shared in this repository contains empty values for all tokens/keys/secrets, meant for any user to fill in with their own credentials. The script then reads the corpus file built by the above and randomly selects a quote/sentence to post to Twitter. It keeps track of what it has posted the previous 10 times to avoid repetition (with certainty) posts over a short/medium period of time by creating and exporting a pickle of an array into your current directory (which it then reads upon subsequent executions). This script is notably dependent on the tweepy, json and numpy modules.
\ No newline at end of file
+The RousseaBot script then reads the corpus file built by the above and randomly selects a quote/sentence to post to Twitter. It keeps track of what it has posted the previous *10 times* within an array to avoid repetition (with certainty) over a short/medium time period by *pickling* this array in your current directory (which it then reads upon subsequent executions).
```
Comment (alamoot) [inline README.md:11]: "within an array" 

Is it important for the reader that you're indexing in an array?

File: README.md [hunk 2]
```diff
@@ -1,9 +1,13 @@
...
+
+Please note that RousseauBot is also dependent on a JSON file containing the credentials to RousseauBot's Twitter account (API/access keys/tokens), which it reads and uses in accessing the account. *A JSON file containing empty values for all tokens/keys/secrets is shared in this repository* for users to fill in with their own credentials to try out the script. To generate/access your own credentials, please visit the Twitter developers portal.
```
Comment (alamoot) [inline README.md:13]: You talk about this empty `JSON` file, might as well write down the name too.

File: corpus.txt [hunk 3]
```diff
@@ -0,0 +1,524 @@
+I enter upon my task without proving the importance of the subject
```
Comment (alamoot) [inline corpus.txt:1]: This hole file is not needed on `git`. Remind me to show you how to go about it using `gitignore`

File: RousseauBot.py [hunk 4]
```diff
@@ -14,63 +14,99 @@
...
     
-api = tw.API(auth)
+    def runAll(self):
+        self.updateDir()
+        self.readCredentials()
+        self.readQuotes()
+        self.readPreviousIndices()
+        self.twitterAuth(self.pickTrackTweet())
+        
+# Define user-updatable constants for files containing credentials, quotes, previous tweets; then run
```
Comment (alamoot) [inline RousseauBot.py:106]: Define these `const`s as part of the `class` itself

File: RousseauBot.py [hunk 5]
```diff
@@ -14,63 +14,99 @@
...
-    pk.dump(prevIndex, indexFile)
-
-quoteToTweet = corpus[index].strip('\n')
-
-# Authenticate credentials with Twitter to access RousseauBot's account
-try:
-    auth = tw.OAuthHandler(apiKey, apiSecret)
-except:
-    sys.exit("API request rejected by Twitter.")
+    # Set current folder to be the working directory
```
Comment (alamoot) [inline RousseauBot.py:30]: See the following for `pythonic` way of method commenting: https://stackoverflow.com/questions/2357230/what-is-the-proper-way-to-comment-functions-in-python

File: RousseauBot.py [hunk 6]
```diff
@@ -14,63 +14,99 @@
...
+        
+        quoteToTweet = self.corpus[index].strip('\n')
+        
+        # Pickle the previous indices list
+        with open(self.previousTweets, 'wb') as indexFile:
+            pk.dump(self.previousIndices, indexFile)
+            
+        return quoteToTweet
+    
+    # Authenticate credentials with Twitter to access RousseauBot's account
```
Comment (alamoot) [inline RousseauBot.py:84]: It's good practice to mention existence (and when they happen) of `exit`s and `exception`s in method comments

File: RousseauBot.py [hunk 7]
```diff
@@ -14,63 +14,99 @@
...
+    def twitterAuth(self, quote):
```
Comment (alamoot) [inline RousseauBot.py:85]: Method names suggest it only authenticates to twitter, but it also posts a tweet

File: RousseauBot.py [hunk 8]
```diff
@@ -14,63 +14,99 @@
...
+        except:
+            self.previousIndices = []
+    
+    
+    def pickTrackTweet(self):
+        
+        # Generate a random number such as we won't index a line from the corpus that was already indexed in the past 10 days
+        index = np.random.randint(0, len(self.corpus)-1)
+        while index in self.previousIndices:
+            index = np.random.randint(0, len(self.corpus)-1)
```
Comment (alamoot) [inline RousseauBot.py:67]: Lines `67` and `65` are duplicates. Can you think of a way of using the line once?

File: RousseauBot.py [hunk 9]
```diff
@@ -14,63 +14,99 @@
...
+    # RousseauBot's Twitter credentials are stored in a json file, read/stored into variables below
+    def readCredentials(self):
+        with open(self.credentialFile, 'r') as file:
+            credDict = json.load(file)
 
-try:
-    auth.set_access_token(accessToken, accessTokSecret)
-except:
-    sys.exit("Authentication failed.")
+        self.apiKey          = credDict['API Key']
```
Comment (alamoot) [inline RousseauBot.py:39]: Should we care if any of these fields in the `credentialFile` are missing?

File: corpusBuilder.py [hunk 10]
```diff
@@ -11,53 +11,77 @@
...
+    
+    def runAll(self):
+        self.dirUpdater()
+        self.HTMLparser()
+        self.corpusWriter()
 
-quotes = []
 
-# These paragraph numbers outline the range of "On the Social Contract" <p> tags
+# Define user-editable constants to be passed into class object as method arguments, then run
```
Comment (alamoot) [inline corpusBuilder.py:80]: `const`s same as last file

File: RousseauBot.py (hunk 8 above)
Comment (bsun94) [reply RousseauBot.py:67]: no do while

**Pull request merged**